### DISCUSSION TITLE
Solving Peridynamics Problem with Retained FE Mesh

### DISCUSSION THREAD

**Discussion by ppandit95** [idaholab/moose#18582] [Q&A Meshing] [answered]
Dear MOOSE Community

I was trying to run a test application with retain_fe_mesh option as "true" but I was not able to solve the problem and ran into error as -
*** ERROR ***
The following error occurred in the object "mesh", of type "PeridynamicsMesh".

Querying node ID exceeds the available PD node IDs!

But was able to vizualise the mesh with '--mesh-only' option while executing . So my querry iswhather can one solve the problem with retaining FE mesh or not . 

Regards 
Pushkar

**Comment by GiudGiud** [comment]
@dschwen not sure who to tag on peridynamics

**Reply by GiudGiud** [reply]
actually, @hchen139

**Comment by hchen139** [comment, marked as answer]
The retained FE mesh is for the purpose of developing other capabilities in the future. For now, we don't really use the retained FE mesh for any PD problems. You will need to set the "retain_fe_mesh" option to false in order to use the current PD capabilities.

BTW, why do you need to retain the FE mesh in your PD simulation?

**Comment by ppandit95** [comment]
Thanks alot for the explanation, ohh regarding retained_fe_mesh  I was just
playing around in order to understand how peridynamics module really works.
